packages/contracts-bedrock/test/properties/halmos/OptimismSuperchainERC20.t.sol

```diff
@@ -0,0 +1,237 @@
+                    && MESSENGER.crossDomainMessageSender() == address(optimismSuperchainERC20)
+            );
+        } catch {
+            assert(
+                _sender != address(MESSENGER)
+                    || MESSENGER.crossDomainMessageSender() != address(optimismSuperchainERC20)
+            );
+        }
+    }
+
+    /// @custom:property-id 8
+    /// @custom:property `sendERC20` with a value of zero does not modify accounting
+    function check_sendERC20ZeroCall(address _from, address _to, uint256 _chainId) public {
+        /* Preconditions */
+        vm.assume(_to != address(0));
+        vm.assume(_chainId != CURRENT_CHAIN_ID);
+        vm.assume(_to != address(Predeploys.CROSS_L2_INBOX) && _to != address(MESSENGER));
+
+        uint256 _totalSupplyBefore = optimismSuperchainERC20.totalSupply();
+        uint256 _fromBalanceBefore = optimismSuperchainERC20.balanceOf(_from);
+
+        vm.startPrank(_from);
+        /* Action */
+        optimismSuperchainERC20.sendERC20(_to, ZERO_AMOUNT, _chainId);
+
+        /* Postcondition */
+        assert(optimismSuperchainERC20.totalSupply() == _totalSupplyBefore);
+        assert(optimismSuperchainERC20.balanceOf(_from) == _fromBalanceBefore);
+    }
+
+    /// @custom:property-id 9
+    /// @custom:property `relayERC20` with a value of zero does not modify accounting
+    function check_relayERC20ZeroCall(address _from, address _to) public {
+        uint256 _totalSupplyBefore = optimismSuperchainERC20.totalSupply();
+        /* Preconditions */
+        uint256 _fromBalanceBefore = optimismSuperchainERC20.balanceOf(_from);
+        uint256 _toBalanceBefore = optimismSuperchainERC20.balanceOf(_to);
+        vm.prank(address(MESSENGER));
+
+        /* Action */
+        optimismSuperchainERC20.relayERC20(_from, _to, ZERO_AMOUNT);
+
+        /* Postconditions */
+        assert(optimismSuperchainERC20.totalSupply() == _totalSupplyBefore);
+        assert(optimismSuperchainERC20.balanceOf(_from) == _fromBalanceBefore);
+        assert(optimismSuperchainERC20.balanceOf(_to) == _toBalanceBefore);
+    }
+
+    /// @custom:property-id 10
+    /// @custom:property `sendERC20` decreases the token's totalSupply in the source chain exactly by the input amount
+    function check_sendERC20DecreasesTotalSupply(
+        address _sender,
+        address _to,
+        uint256 _amount,
+        uint256 _chainId
+    )
+        public
+    {
+        /* Preconditions */
+        vm.assume(_to != address(0));
+        vm.assume(_chainId != CURRENT_CHAIN_ID);
+
+        vm.prank(Predeploys.L2_STANDARD_BRIDGE);
+        optimismSuperchainERC20.mint(_sender, _amount);
+
+        uint256 _totalSupplyBefore = optimismSuperchainERC20.totalSupply();
+        uint256 _balanceBefore = optimismSuperchainERC20.balanceOf(_sender);
+
+        vm.prank(_sender);
+        /* Action */
+        optimismSuperchainERC20.sendERC20(Predeploys.CROSS_L2_INBOX, _amount, _chainId);
```

Comment:
Why send it to the cross_l2_inbox instead of `to`?

side question: would that revert on relay on a live system when trying to relay?